### ISSUE TITLE
Docker Setup - User was denied access on the database .public

### ISSUE THREAD

**Issue by IrealiTY** [ghostfolio/ghostfolio#1049]
Tried a few variations of the setup and all failed with this error:

```
ghostfolio_1  | (node:1) UnhandledPromiseRejectionWarning: Error: 
ghostfolio_1  | Invalid `this.prismaService.authDevice.findUnique()` invocation in
ghostfolio_1  | /ghostfolio/apps/api/main.js:1:28152
ghostfolio_1  | 
ghostfolio_1  | → 1 (()=>
ghostfolio_1  |   User `` was denied access on the database `.public`
ghostfolio_1  |     at Object.request (/ghostfolio/apps/api/node_modules/@prisma/client/runtime/index.js:45584:15)
ghostfolio_1  |     at async PrismaService._request (/ghostfolio/apps/api/node_modules/@prisma/client/runtime/index.js:46405:18)
ghostfolio_1  | (Use `node --trace-warnings ...` to show where the warning was created)
ghostfolio_1  | (node:1) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
```

the only change I made was to add the `ALPHA_VANTAGE_API_KEY` and set randomized passwords for Redis, Postgres, JWT_SECRET_KEY and ACCESS_TOKEN_SALT

This error is happens on the initial
`docker-compose --env-file ./.env -f docker/docker-compose.yml up -d` (if I check the logs of the docker container)
and gets prompted again on 
`docker-compose --env-file ./.env -f docker/docker-compose.yml exec ghostfolio yarn database:setup`

**Comment by IrealiTY** [comment]
Tried that too, started Redis, Postgres and the app one by one (checked for "Ready to accept connections" in the log) but the app errored out anyway

https://zerobin.net/?146c7f44b253b5f9#+ym4PktBBBQfzcwEOLyPaURIAVSjckkIfsYf5jc14Vw=

**Comment by dtslvr** [comment]
> ghostfolio_1  |   User `` was denied access on the database `.public`

It looks like your user (empty?) does not have access to the database and the schema `public`. Can you verify @IrealiTY? Someone with a similar [issue](https://github.com/prisma/prisma/issues/12100#issuecomment-1057412562) has solved it by using the root user.

**Comment by IrealiTY** [comment]
I didn't get around to test it just yet but I I noticed when relaunching the container that it the logs drop
```
WARNING: The POSTGRES_USER variable is not set. Defaulting to a blank string.
WARNING: The POSTGRES_PASSWORD variable is not set. Defaulting to a blank string.
WARNING: The POSTGRES_DB variable is not set. Defaulting to a blank string.
WARNING: The REDIS_PASSWORD variable is not set. Defaulting to a blank string.
```

which could be related? those vars are set in the .env file though

I use `sudo docker-compose --env-file ./.env -f docker/docker-compose.yml up -d`  from the main folder
It was seemingly not using the .env variables.

After copying the .env into the docker folder and just launching docker-compose it now uses the correct usernames but still spits out the error but just saying that the provided user doesn't have access

```
ghostfolio_1  | (node:1) UnhandledPromiseRejectionWarning: Error: 
ghostfolio_1  | Invalid `this.prismaService.authDevice.findUnique()` invocation in
ghostfolio_1  | /ghostfolio/apps/api/main.js:1:28152
ghostfolio_1  | 
ghostfolio_1  | → 1 (()=>
ghostfolio_1  |   Authentication failed against database server at `postgres`, the provided database credentials for `user` are not valid.
ghostfolio_1  | 
ghostfolio_1  | Please make sure to provide valid database credentials for the database server at `postgres`.
ghostfolio_1  |     at Object.request (/ghostfolio/apps/api/node_modules/@prisma/client/runtime/index.js:45584:15)
ghostfolio_1  |     at async PrismaService._request (/ghostfolio/apps/api/node_modules/@prisma/client/runtime/index.js:46405:18)
ghostfolio_1  | (Use `node --trace-warnings ...` to show where the warning was created)
ghostfolio_1  | (node:1) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
```


# Got it working
Before sending this post I tried a bit around and I think I found the issue?
Seems like there are conflicting variables/docker not fully reading the .env file and fighting with itself? Not sure, not a Docker expert 😄 

1. moved .env file to docker folder
2. modified docker-compose file like that:

```yaml
version: '3.7'
services:
  ghostfolio:
    image: ghostfolio/ghostfolio:latest
    environment:
      DATABASE_URL: ${DATABASE_URL}
      REDIS_HOST: 'redis'
      REDIS_PASSWORD: ${REDIS_PASSWORD}
      JWT_SECRET_KEY: ${JWT_SECRET_KEY}
      ACCESS_TOKEN_SALT: ${ACCESS_TOKEN_SALT}
      ALPHA_VANTAGE_API_KEY: ${ALPHA_VANTAGE_API_KEY}
    ports:
      - 3333:3333

  postgres:
    image: postgres:12
    volumes:
      - postgres:/var/lib/postgresql/data

  redis:
    image: 'redis:alpine'

volumes:
  postgres:

```
3. modified the .env file as such:
```config
COMPOSE_PROJECT_NAME=ghostfolio-development

# CACHE
REDIS_HOST=localhost
REDIS_PORT=6379
REDIS_PASSWORD=COOL_PASSSWORD_HERE

# POSTGRES
POSTGRES_DB=ghostfolio-db
POSTGRES_USER=user
POSTGRES_PASSWORD=COOL_PASSSWORD_HERE

ACCESS_TOKEN_SALT=COOL_PASSSWORD_HERE
ALPHA_VANTAGE_API_KEY=COOL_PASSSWORD_HERE
DATABASE_URL=postgresql://${POSTGRES_USER}:${POSTGRES_PASSWORD}@postgres:5432/${POSTGRES_DB}?sslmode=prefer
JWT_SECRET_KEY=COOL_PASSSWORD_HERE
```

4. launched docker-compose directly from the docker folder `sudo docker-compose down && sudo docker-compose up -d --force-recreate` (used recreate since I had already a faulty instance running)
5.  ran the database setup command also from inside the docker folder `sudo docker-compose exec ghostfolio yarn database:setup`
6. result : 😄 
![image](https://user-images.githubusercontent.com/1223686/176288518-2575c74c-1add-425d-ba9f-99a2ec7f36fc.png)

TL:DR - something about variables not being passed on correctly?

**Comment by Petemir** [comment]
I have the same problem -- if the env file is in a folder different from the compose file, it appears that the command does not read the environment variables appropriately. I do not have this problem with other containers.
